rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactoryBuilder.java

```diff
@@ -0,0 +1,351 @@
+                SecuritySettingsService service = bundleContext.getService(serviceReference);
+                if (service != null) {
+                    return service;
+                }
+            }
+        }
+        throw new SecurityServiceException("Could not get SecuritySettingsService");
+    }
+
+    private STSClientConfiguration getStsClientConfiguration() throws SecurityServiceException {
+        BundleContext bundleContext = FrameworkUtil.getBundle(this.getClass()).getBundleContext();
+        if (bundleContext != null) {
+            ServiceReference<STSClientConfiguration> serviceReference = bundleContext
+                    .getServiceReference(STSClientConfiguration.class);
+            if (serviceReference != null) {
+                STSClientConfiguration service = bundleContext.getService(serviceReference);
+                if (service != null) {
+                    return service;
+                }
+            }
+        }
+        throw new SecurityServiceException("Could not get STSClientConfiguration");
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass)
+            throws SecurityServiceException {
+        return buildFactory(endpointUrl, interfaceClass, null, null);
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass,
+            String username, String password) throws SecurityServiceException {
+        return buildFactory(endpointUrl, interfaceClass, username, password, null, false);
+    }
+
+    /**
+     * Creates a factory that will return security-aware clients.
+     *
+     * @see SecureCxfClientFactoryBuilder.SecureCxfClientFactory
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass,
+            String username, String password, List<?> providers, boolean disableCnCheck)
+            throws SecurityServiceException {
+        if (StringUtils.isEmpty(endpointUrl) || interfaceClass == null) {
+            throw new IllegalArgumentException(
+                    "Called without a valid URL, will not be able to connect.");
+        }
+
+        JAXRSClientFactoryBean jaxrsClientFactoryBean = new JAXRSClientFactoryBean();
+        jaxrsClientFactoryBean.setServiceClass(interfaceClass);
+        jaxrsClientFactoryBean.setAddress(endpointUrl);
+        jaxrsClientFactoryBean.setClassLoader(interfaceClass.getClassLoader());
+        jaxrsClientFactoryBean.getInInterceptors().add(new LoggingInInterceptor());
+        jaxrsClientFactoryBean.getOutInterceptors().add(new LoggingOutInterceptor());
+
+        if (!CollectionUtils.isEmpty(providers)) {
```

Comment:
Could if it was the real CollectionUtils. That  one is embedded in cxf.common for whatever weird reason. I looked at the code and it's basically the same thing though. So...it would be an additional dependency just to make something look slightly cleaner. In my opinion this is better, since it's already there in a dependency we need anyways.